pkg/controllers/dynakube/apimonitoring/reconciler.go

```diff
@@ -75,6 +75,8 @@ func (r *Reconciler) createObjectIdIfNotExists(ctx context.Context) (string, err
 		return "", errors.WithMessage(err, "error creating dynatrace settings object")
 	}
 
+	r.dk.Status.KubernetesClusterMEID = meID
```

Comment:
you could simplify the logic above a bit, you are basically doing a monitored-entity reconcile, that is unnecessary if we already know the MEID.

you can just have a monitored-entity reconciler at the beginning as well, it will not run if we already know the MEID, so it will not conflict with any other run of the monitored-entity reconciler, and you can simplify the code here.

(also you can get the kubeSystemUUID from the dynakube)

Reply:
(hopefully) done in https://github.com/Dynatrace/dynatrace-operator/pull/3447/commits/3459e905e7dbe1da30717fe4156a47e3064ace70